build-images.sh

```diff
@@ -9,50 +9,63 @@ function usage() {
     exit 0
 }
 
-while [ -n "$1" ]; do
-    case $1 in
-    -h | --help) usage ;;
-    -?*)
-        echo "Unknown option '$1'" 1>&2
-        exit 1
+parse_args() {
+  local SHORT=h,d:
+  local LONG=distro:
+
+  local OPTS=$(getopt -o $SHORT -l $LONG --name "$0" -- "$@")
+
+  if [ $? != 0 ] ; then echo "Failed to parse options...exiting." >&2 ; exit 1 ; fi
+
+  eval set -- "$OPTS"
+
+  . settings.sh  # set initial values
+
+  while true ; do
+    case "$1" in
+      -d | --distro )
+        ROS_DISTRO="$2"
+        shift 2
```

Comment:
I think I prefer to have the distro as a positional required argument (as it was before) to assure the right image is built. By the way, I think the scripts `run-tests.sh` and `run-devel.sh` should have the same behavior

Reply:
But the DISTRO before wasn't optional?

Reply:
It wasn't optional for the `build-images.sh` script. I think it could remain required for this script and optional for the test and devel scripts.